### ISSUE TITLE
Let user download videos on wifi so they can watch later

### ISSUE THREAD

**Issue by robd003** [lbryio/odysee-ios#198]
It would be really cool to let users download videos to their device so they could watch them later when they don't have a cell signal or to reduce bandwidth on cell plans.

**Comment by tsheaff** [comment]
Agreed. If users want to do this on cellular connection I don't think the app should stop them either, perhaps just a little confirmation prompt (or make it a global setting toggle). A lot of people have unlimited cell plans and will just want to download videos before boarding an airplane or something so we shouldn't block cellular IMO.

**Comment by tsheaff** [comment]
And then if you open the app without a network connection at all (e.g. under a tunnel, or phone is on airplane etc) then it defaults to a view of all of your downloaded videos only, perhaps in just recency-of-download order.

**Comment by anohren** [comment]
I would appreciate this. 

There's also a lightweight version of this feature that could potentially require less housekeeping for the user:

Often I want to download a single video right before I leave an area with WiFi, in order to finish listening to it while walking somewhere else. In those cases I'd appreciate just a menu option like "Buffer entire video" and a clear buffer progress bar which means I could leave as soon as it's all (or enough of it) buffered. YouTube has pretty aggressive buffering which works well in this use case. 

The benefit would be that I'd not have to think about deleting it later since it's inherently transient. I also wouldn't have to navigate away into a separate Downloads view, restart the video and find the place where I left off just to continue watching. 

Or maybe it only complicates things, since you'd have to involve persistence into the buffering strategy to take large files into account. The original suggestion plus a "delete when watched" preference with an "after 24 hours" option might be more straightforward. It'd just have been nice with the seamlessness.
